tasks/main.yml

```diff
@@ -661,17 +661,33 @@
       {{ ad_integration_user }}@{{ ad_integration_realm | upper }}
   when: mssql_ad_configure | bool
   block:
-    - name: Join to realm {{ ad_integration_realm }}
-      include_role:
-        name: fedora.linux_system_roles.ad_integration
+    - name: Verify that required variables are provided
+      assert:
+        that: ad_integration_realm is defined
 
-    - name: Ensure the adutil package
+    - name: Verify if the mssql_ad_obtain_kerberos var is provided correctly
+      when: mssql_ad_obtain_kerberos | bool
+      assert:
+        that:
+          - ad_integration_user is defined
+          - ad_integration_password is defined
+        fail_msg: >-
+          You have mssql_ad_obtain_kerberos set to true.
+          You must set ad_integration_user and ad_integration_password variables
+          to obtain Kerberos ticket.
+
+    - name: Ensure adutil
       package:
         name: adutil
         state: present
 
-    - name: Create keytab file
-      when: mssql_ad_keytab_file is none
+    - name: Join to realm {{ ad_integration_realm }}
+      include_role:
+        name: fedora.linux_system_roles.ad_integration
+      when: mssql_ad_join | bool
+
+    - name: Obtain Kerberos ticket
+      when: mssql_ad_obtain_kerberos
```

Comment:
You still need a kerberos ticket if you want to be able to run any `adutil` commands later on. Another option:
If `ad_integration_user` is specified, run the same block and then also check if the user exists, try to create it if it doesn't, and set the SPN. Then, regardless of whether `ad_integration_user` was specified, get a kerberos ticket for `mssql_ad_sql_user_name` and then get the keytab using that account.

That way, you know a ticket was generated before trying to create a keytab file. This would also verify `mssql_ad_sql_password`

Reply:
My intent was to let users who like to obtain kerberos themselves can do this. By default, `mssql_ad_obtain_kerberos` is `true`, and I expect that the majority would keep it `true` and just provide `ad_integration_user`. I actually do not see an issue in obtaining kerberos ticket even if you already have it obtained, the role is idempotent and will just skip obtaining if the ticket is activated.

What's your intent? Maybe to let users choose for what username to obtain a kerberos ticket?

Reply:
> Do you want to allow for situations where 3 accounts are used? 1 for AD joining, 1 for sqluser creationg, and sqluser.

I'd be happy to allow this if this is required. 

1. Account for AD joining is `ad_integration_user` - not applicable when `mssql_ad_join: false`
2. Account for sqluser creation is the one that obtains kerberos ticket. This would be either `ad_integration_user`, if provided, and if `ad_integration_user` is not provided, then `mssql_ad_sql_user_name`.
   Note that when `mssql_ad_join: true`, `ad_integration_user` will always be provided.  And if `ad_integration_user` is not defined, the `mssql_ad_kerberos_username` user must be pre-created in AD.

  The role can introduce another variable, like `mssql_ad_kerberos_username`, for the case when users want to join to AD and provide `ad_integration_user`, but want to use e.g. `mssql_ad_sql_user_name` or other user to obtain kerberos and create keytab.
3. `mssql_ad_sql_user_name`
Does it make sense?

Reply:
I ended up adding `mssql_ad_kerberos_user` and `mssql_ad_kerberos_password`, I think this is a needed flexibility. And I removed mssql_ad_obtain_kerberos, the role can ensure it for users, and in the case when users have a third AD user that they want to use for sqluser  and keytab creation, the can use the mssql_ad_kerberos_user variables.

Reply:
I was actually saying 3 accounts might be a little excessive. I was only pointing out that having 3 accounts is the only situation I could see where having `mssql_ad_obtain_kerberos` not reflect whether `ad_integration_user` is set would be beneficial. If you want to have that flexibility in there, then I can completely agree with how each account would be used and the default values as you described them.

Reply:
Haha, I didn't read the irony in your commit. But the resulted in a good thing.
IMO the logic of the default value of `mssql_ad_kerberos_user` is not exactly clear, maybe some users would want to set `mssql_ad_kerberos_user` to an actual username, whether it be the same as `ad_integration_user` or `mssql_ad_sql_user_name`, just to be explicit.

Reply:
More flexibility is rarely a bad thing. Since it's already written, might as well keep it. It just means it can be used in more environments and it doesn't really complicate the code. It just adds a variable.

Reply:
IMO it's good that the role always ensures that Kerberos ticket exists when a pre-generated `mssql_ad_keytab_file` file is provided. This is a simple step to automate, so I find the  `mssql_ad_obtain_kerberos` variable redundant.